### ISSUE TITLE
Code needs updating as gates in Aqua were migrated to Terra

### ISSUE THREAD

**Issue by woodsp-ibm** [qiskit-community/qiskit-textbook#390]
content/ch-algorithms/quantum-counting.ipynb has the following
```
from qiskit.aqua.circuits.gates.multi_control_toffoli_gate import _cccx
```
The gates that were in Aqua were migrated over to Terra so this needs fixing. Also I will note that it used a method marked private which is not good practice.

**Comment by Sanjay-995** [comment]
I would also like to add the HHL Algorithm import statements in the Qiskit textbook, 
from qiskit.aqua import run_algorithm
from qiskit.aqua.input import LinearSystemInput
from qiskit.quantum_info import state_fidelity
from qiskit.aqua.algorithms.classical import ExactLSsolver
import numpy as np
ImportError: cannot import name 'run_algorithm' from 'qiskit.aqua' (/opt/conda/lib/python3.7/site-packages/qiskit/aqua/__init__.py)
I guess the modules have been moved and need to be updated accordingly.

**Comment by frankharkins** [comment]
The initial problem has been fixed in f299364b2c7d5e244004879f03a5fd965ba0f831, still working on the HHL chapter.
